core/src/jvmMain/kotlin/com/xebia/functional/xef/conversation/PlatformConversation.jvm.kt

```diff
@@ -58,12 +60,14 @@ actual constructor(
       }
       .asCompletableFuture()
 
-  fun chatFunction(target: Class<*>): CFunction =
-    CFunction(
+  fun chatFunction(target: Class<*>): CFunction {
+    val targetString = JacksonSerialization.schemaGenerator.generateSchema(target).toString()
+    return CFunction(
       name = target.simpleName,
       description = "Generated function for ${target.simpleName}",
-      parameters = JacksonSerialization.schemaGenerator.generateSchema(target).toString()
+      parameters = Json.parseToJsonElement(targetString).jsonObject
```

Comment:
The current implementation (generating schema with Jackson, print to JSON string, and parsing with Kotlinx serialization makes me uncomfortable 😅), but at this point I think it's the right way.

Reply:
Yes, I feel the same, but as Raul mentioned to me, I think this will change in a future PR.